network/src/testutils/fake_socket.rs

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//!
+//! This module exposes two types of sockets useful for tests:
+// - ExposingTestSocket: a wrapper around MemorySocket to retrieve handshake messages being sent.
+// - ReadOnlyTestSocket: a socket that can be read from in different ways.
+//!
+//! What is this fake socket for?
+//!
+//! - does nothing on a write
+//! - read behaves according to mode
+//!
+//!
+//! used to quickly consume fuzzing data and dismiss socket writes.
+//! ```
+//!
+//! ```
+
+use futures::{
+    io::{AsyncRead, AsyncWrite},
+    ready,
+    task::{Context, Poll},
+};
+use memsocket::MemorySocket;
+use std::{io, pin::Pin};
+
+//
+// ReadOnlyTestSocket
+// ==================
+//
+
+pub struct ReadOnlyTestSocket<'a> {
+    /// the content
+    content: &'a [u8],
+    /// the socket will read byte-by-byte
+    fragmented: bool,
+    /// continue to read 0s once content has been fully read
+    trailing: bool,
+}
+
+impl<'a> ReadOnlyTestSocket<'a> {
+    pub fn new(content: &'a [u8]) -> Self {
+        Self {
+            content,
+            fragmented: false,
+            trailing: false,
+        }
+    }
+
+    /// reads will have to be done byte by byte
+    #[allow(dead_code)]
```

Comment:
Why are these dead code?

Reply:
ah! I figured out why I was getting linting errors. I forgot to put the testutils module under a `#[cfg(test)]`